libsolidity/ast/Types.cpp

```diff
@@ -406,28 +603,52 @@ bool IntegerConstantType::isImplicitlyConvertibleTo(Type const& _convertTo) cons
 			return true;
 		return false;
 	}
+	else if (_convertTo.category() == Category::FixedPoint)
+	{
+		auto targetType = dynamic_cast<FixedPointType const*>(&_convertTo);
+		if (m_value == 0)
+			return true;
+		int forSignBit = (targetType->isSigned() ? 1 : 0);
+		if (m_value > 0)
+		{
+			bool properlyScaledBits = m_scalingFactor <= targetType->fractionalBits() ? 
```

Comment:
thinking about this...this should actually be checking to see whether the whole numbers of the rational can fit in the integer bits and if the remainder can fit in the fractional bits...Otherwise we haven't properly converted, have we?